File: 3rd Quarter/Data Structures & Algorithms/Programs/Unit3/Maze/Point2D.cpp
#include "Point2D.h"

int Point2D::count = 0;

Point2D::Point2D(int x, int y)
{
	this->x = x;
	this->y = y;
	count++;
}

Point2D::~Point2D() { count--; }

void Point2D::setX(int x) { this->x = x; }

void Point2D::setY(int y) { this->y = y; }

int Point2D::getX() const { return this->x; }

int Point2D::getY() const { return this->y; }

Point2D& Point2D::operator+(const Point2D rhs)const
{
	return Point2D(this->getX() + rhs.getX(),
		this->getY() + rhs.getY());
}

Point2D& Point2D::operator-(const Point2D rhs)const
{
	return Point2D(this->getX() - rhs.getX(),
		this->getY() - rhs.getY());
}

File: 2nd Quarter/Obj-Oriented Programming/Programs/Vectors/Point2D.java

package vectors;

/**
 * Creates a point in 2D space
 * @author djc35
 */
public class Point2D 
{
    protected int x;
    protected int y;
	
	
	public Point2D() 
        {
		this.x = 0;
		this.y = 0;
	}
	
	
	public Point2D(int xy) 
        {
		this.x = xy;
		this.y = xy;
	}
	
	
	public Point2D(int x, int y) 
        {
		this.x = x;
		this.y = y;
	}
	
	
	public void setX(int x) {this.x = x;}
	public void setY(int y) {this.y = y;}
	
	
	public int getX() {return x;}
	public int getY() {return y;}
	
	
	public boolean Equals(Object object) 
        {
		
		if(object instanceof Point2D) 
                {
			
			Point2D p = (Point2D)object;
			if( this.getX() == p.getX() && this.getY() == p.getY() )
				return true;		
		}
		return false;
		
	}
	
	
	public String toString() {return "(" + this.x + ", " + this.y + ")";}
}
